src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -321,6 +344,7 @@ public Visitable visit(PackageDeclaration node, Void arg) {
 
   @Override
   public Visitable visit(ClassOrInterfaceDeclaration node, Void arg) {
+    setDeclaredMethod(node.getMethods());
```

Comment:
I'm not sure that this is sufficient, because I think the set isn't properly cleared when a class body is exited. The proper abstraction here is probably a stack of sets, similar to the standard algorithm for tracking scopes.

Please add a test case where there are two or more inner classes that demonstrates this issue. If I'm correct, the set of declared methods that are available in the second inner class will still include any methods that are declared in the first inner class. You'll need to make sure that the second inner class uses an unsolved method from an unsolved superclass or superinterface that has the same name/signature as one of the methods that the first inner class declares to show the bug.